src/generated/asyncResult.generated.ts

```diff
@@ -0,0 +1,417 @@
+// ========================================================
```

Comment:
I'm not sure why this needs to be generated, but I trust you that it's necessary. My question for you is, similarly to the generated common JS modules, can you not version this in Git and only version their `generator` counterpart to prevent mistakes?

Reply:
Yes, i can add the generated code files to gitignore, since they are only necessary when building.